src/app/utils/utils.ts

```diff
@@ -29,17 +29,17 @@ export function getParentUrl(router: Router): string {
  * @param array A simple array to be grouped.
  * @param propertyToGroupBy The key of the property in the elements by whose value the result is grouped.
  */
-export function groupBy<T>(
-  array: T[],
-  propertyToGroupBy: keyof T
-): Map<string, T[]> {
+export function groupBy<ENTITY, PROPERTY extends keyof ENTITY>(
+  array: ENTITY[],
+  propertyToGroupBy: PROPERTY
```

Comment:
Why does PROPERTY need to be a generic var itself? what is the difference to the previous `keyof T` as type of propertyToGroupBy?

Reply:
This allows to reference the correct (same) type for the return value. Because `PROPERTY` always refers to the property of the input value, whereas `keyof ENTITY` could be a different one than the input property.
E.g. when using `groubBy(<array of notes>, "date") the return type will be `Map<Date, Note[]>`.